modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,846 @@
+        mc.setRelatesTo(reference);
+    }
+
+    public EndpointReference getReplyTo() {
+        return mc.getReplyTo();
+    }
+
+    public void setReplyTo(EndpointReference reference) {
+        mc.setReplyTo(reference);
+    }
+
+    public EndpointReference getTo() {
+        return mc.getTo();
+    }
+
+    public void setTo(EndpointReference reference) {
+        mc.setTo(reference);
+    }
+
+    public void setWSAAction(String actionURI) {
+        mc.setWSAAction(actionURI);
+    }
+
+    public String getWSAAction() {
+        return mc.getWSAAction();
+    }
+
+    public String getSoapAction() {
+        return mc.getSoapAction();
+    }
+
+    public void setSoapAction(String string) {
+        mc.setSoapAction(string);
+    }
+
+    public void setWSAMessageID(String messageID) {
+        mc.setWSAMessageID(messageID);
+    }
+
+    public String getWSAMessageID() {
+        return mc.getWSAMessageID();
+    }
+
+    public boolean isDoingMTOM() {
+        return mc.isDoingMTOM();
+    }
+
+    public boolean isDoingSWA() {
+        return mc.isDoingSWA();
+    }
+
+    public void setDoingMTOM(boolean b) {
+        mc.setDoingMTOM(b);
+    }
+
+    public void setDoingSWA(boolean b) {
+        mc.setDoingSWA(b);
+    }
+
+    public boolean isDoingPOX() {
+        return mc.isDoingPOX();
+    }
+
+    public void setDoingPOX(boolean b) {
+        mc.setDoingPOX(b);
+    }
+
+    public boolean isDoingGET() {
+        return mc.isDoingGET();
+    }
+
+    public void setDoingGET(boolean b) {
+        mc.setDoingGET(b);
+    }
+
+    public boolean isSOAP11() {
+        return mc.isSOAP11();
+    }
+
+    public void setResponse(boolean b) {
+        mc.setResponse(b);
+    }
+
+    public boolean isResponse() {
+        return mc.isResponse();
+    }
+
+    public void setFaultResponse(boolean b) {
+        mc.setFaultResponse(b);
+    }
+
+    public boolean isFaultResponse() {
+        return mc.isFaultResponse();
+    }
+
+    public int getTracingState() {
+        return mc.getTracingState();
+    }
+
+    public void setTracingState(int tracingState) {
+        mc.setTracingState(tracingState);
+    }
+
+    public Stack<FaultHandler> getFaultStack() {
+        return mc.getFaultStack();
+    }
+
+    public void pushFaultHandler(FaultHandler fault) {
+        mc.pushFaultHandler(fault);
+    }
+
+    public void pushContinuationState(ContinuationState continuationState) {
+    }
+
+    public Stack<ContinuationState> getContinuationStateStack() {
+        return null;
+    }
+
+    public boolean isContinuationEnabled() {
+        return false;
+    }
+
+    public void setContinuationEnabled(boolean contStateStackEnabled) {
+    }
+
+    public Log getServiceLog() {
+        return LogFactory.getLog(CommonScriptMessageContext.class);
+    }
+
+    public Mediator getSequenceTemplate(String key) {
+        return mc.getSequenceTemplate(key);
+    }
+
+    private String serializeJSON(Object obj) {
```

Comment:
Shall we move this function to Util class? This may useful for other features in future